### ISSUE TITLE
props or state not coming through

### ISSUE THREAD

**Issue by lhtdesignde** [s-yadav/react-number-format#4]
Hey,
great library. work good for me just one little thing:
I dynamically disable the input field and it seems like the component is not applying the property although it's updating. Is it possible to get that working somehow?

```
            <NumberFormat 
                          id='card-number'
                          displayType={'tel'}
                          format='#### #### #### ####'
                          disabled={this.props.submitting} />
```
"submitting" is a boolean property that gets updated when fetching.

**Comment by s-yadav** [comment]
Strange it should work, as I am adding all the props to input element.
https://github.com/s-yadav/react-number-format/blob/master/src/number_format.js#L148

Check if `this.props.submitting` is giving expected value.

Btw  valid values  for displayType are 'text' or 'nput'

**Comment by lhtdesignde** [comment]
yes this.props.submitting is working as expected. if I don't use the component, just a normal input it works. changed the input type now to input but this also didn't help.
I'm using also a spread operator on the NumberFormat. could that overwrite something somehow?

```
            <NumberFormat {...cardNumber}
                          id='card-number'
                          displayType={'input'}
                          format='#### #### #### ####'
                          autoComplete='cc-number'
                          disabled={submitting}
                          aria-disabled={submitting} />
```

**Comment by s-yadav** [comment]
Can you setup a codepen/jsfiddle ?

**Comment by lhtdesignde** [comment]
it's a bit more complicated because also redux-form is in the middle. it seems like "false" is coming through initially but then it's not updating when the prop gets updated. not sure if i can setup a codepen/jsfiddle for the whole case. will try.

**Comment by s-yadav** [comment]
Ok I tried in example codepen http://codepen.io/anon/pen/rrPgox and it does work.
```js
<NumberFormat disabled={true} format="#### #### #### ####" />
<NumberFormat {...{disabled:false}} disabled={true} format="#### #### #### ####" />
```

**Comment by s-yadav** [comment]
Looks like I got the issue,

https://github.com/s-yadav/react-number-format/blob/master/src/number_format.js#L20
```js
shouldComponentUpdate:function(newProps){
    return !(newProps!==this.props && newProps.value === this.props.value);
  },
```

I will fix and update this.

**Comment by lhtdesignde** [comment]
cool that would be awesome!

**Comment by s-yadav** [comment]
Fixed on 0.1.3.  Let me know if it still exist.

**Comment by lhtdesignde** [comment]
great work! it works now for me. 👍
